docs/basics/101-145-hooks.rst

```diff
@@ -0,0 +1,162 @@
+
+  .. code-block:: bash
+
+     {"type": ["in", ["file", "directory"]], "action": "get", "status": "notneeded"}
+
+  This translates to:  "Match a "not-needed" after :command:`datalad get` of a file or directory."
+  Another example is this::
+
+     '{"type":"dataset","action":"install","status":["eq", "ok"]}'
+
+  which translates to: "Match a successful installation of a dataset".
+
+- The value for ``datalad.result-hook.<name>.call-json`` is specified in its
+  Python notation, and its options are specified as a JSON-encoded dictionary
+  with keyword arguments. Conveniently, a number of string substitutions are
+  supported: a ``dsarg`` argument expands to the ``dataset`` given to the initial
+  command the hook operates in, and any key from the result evaluation can be
+  expanded to the respective value in the result dictionary. Curly braces need to
+  be escaped by doubling them.
+  This is not the easiest specification there is, but its also not as hard as it
+  may sound. Here is how this could look like for a :command:`datalad unlock`::
+
+     $ unlock {{"dataset": "{dsarg}", "path": "{path}"}}
+
+  This translates to "unlock the path the previous command operated on, in the
+  dataset the previous command operated on". Another example is this run command::
+
+     $ run  {{"cmd": "touch {path}_annoyed", "dataset": "{dsarg}", "explicit": true}}
+
+  This translate to "execute a run command in the dataset the previous command operated
+  on. It should create an empty file under the same path the previous command
+  operated on, with an added '_annoyed' in the file name." A final example is this::
+
+     $ run_procedure {{"dataset":"{path}","spec":"cfg_metadatatypes bids"}}
+
+  This hook will run the procedure ``cfg_metadatatypes`` with the argument ``bids``
+  and thus set the standard metadata extractor to be bids.
+
+
+As these variables are configuration variables, they can be set via :command:`git config` [#f2]_::
+
+   $ git config -f .datalad/config --add datalad.result-hook.annoy.call-json 'run {{"cmd":"touch {path}_annoyed", "dataset":"{dsarg}","explicit":true}}'
+   $ git config -f .datalad/config --add datalad.result-hook.annoy.match-json '{"type":["in", ["file"]],"action":"get","status":"notneeded"}'
+
+Here is what this writes to the ``.datalad/config`` file::
+
+    [datalad "result-hook.annoy"]
+        call-json = run {{\"cmd\":\"touch {path}_annoyed\", \"dataset\":\"{dsarg}\",\"explicit\":true}}
+        match-json = {\"type\":[\"in\", [\"file\"]],\"action\":\"get\",\"status\":\"notneeded\"}
+
+Given this configuration in the ``.datalad/config`` file of your dataset, the
+"annoy" hook would be executed whenever you run :command:`datalad get` on a file
+and the command evaluates to "notneeded". The annoy hook would then automatically
+create an empty file with the same name as the one you attempted to get, but with
+an appened ``_annoy`` in the file name [#f3]_.
```

Comment:
This is not possible due to the implied security risk. We do not read .datalad/config for result hooks, but local and global git config only. Otherwise a `datalad update` from an external source could place arbitrary code on a system that is then triggered and executed by datalad.

Reply:
Thanks for the clarification. This also means that the above ``git config`` command is wrong with ``-f .datalad/config``, correct?

Reply:
It is wrong, correct ;-)